modules/planning/tasks/deciders/lane_change_decider/lane_change_decider.cc

```diff
@@ -188,37 +192,48 @@ void LaneChangeDecider::UpdateStatus(double timestamp,
 }
 
 void LaneChangeDecider::PrioritizeChangeLane(
+    const bool is_prioritize_change_lane,
     std::list<ReferenceLineInfo>* reference_line_info) const {
   if (reference_line_info->empty()) {
     AERROR << "Reference line info empty";
     return;
   }
+
   auto iter = reference_line_info->begin();
   while (iter != reference_line_info->end()) {
-    if (iter->IsChangeLanePath()) {
-      reference_line_info->splice(reference_line_info->begin(),
-                                  *reference_line_info, iter);
+    ADEBUG << "iter->IsChangeLanePath(): " << iter->IsChangeLanePath();
+    /* is_prioritize_change_lane == true: prioritize change_lane_reference_line
+       is_prioritize_change_lane == false: prioritize
+       non_change_lane_reference_line */
+    if (is_prioritize_change_lane == iter->IsChangeLanePath()) {
+      ADEBUG << "is_prioritize_change_lane: " << is_prioritize_change_lane;
+      ADEBUG << "iter->IsChangeLanePath(): " << iter->IsChangeLanePath();
       break;
     }
     ++iter;
   }
+  reference_line_info->splice(reference_line_info->begin(),
```

Comment:
Please also double-check what influence on the list iteration order in lane_follow_stage will be caused by the splice operation here. Originally, there was no influence because the lane_change reference line always appears at the front of the list for some reason and the order was never changed. However, after your modification, the order of the reference_line list could be changed while somewhere else in the program is doing the iteration on the list. https://github.com/ApolloAuto/apollo/blob/f2fa04cc578a4b75785f50fbbd9c1c30b0804885/modules/planning/scenarios/lane_follow/lane_follow_stage.cc#L118

Reply:
Fixed